chromaui/chromatic-e2e
# Enable manual snapshots without automatic snapshots

**_This is the non-fork version of #134 (thanks to @jackleslie and @ethriel3695 for the fix!), so that we get CI checks before merging._** 

Issue: https://github.com/chromaui/chromatic-e2e/issues/131

## What Changed

From documentation:

> When `true`, will disable the snapshot that happens automatically at the end of a test when using the Chromatic test fixture.

I believe currently when `disableAutoSnapshot` is set to `true` we are not proceeding to the `prepareArchives` task, so rather than disabling the automatic snapshot we are disabling all snapshots

_**I also disabled Codacy's duplication rule for test directories.**_

## How to test

2. Set up a Cypress project on Chromatic (following [these instructions](https://www.chromatic.com/docs/cypress/)) -- make sure you're using the canary release from this PR
1. set `disableAutoSnapshot: true` globally
3. [Run Cypress](https://www.chromatic.com/docs/cypress/#4-run-cypress)
4. Verify the `cypress/downloads/chromatic-archives` directory has an empty storybook
5. Inside a test, [call](https://www.chromatic.com/docs/cypress/targeted-snapshots/) `cy.takeSnapshot()`
6. Run Cypress again
7. Verify the `cypress/downloads/chromatic-archives` is populated and has a non-empty storybook
8. [Run Chromatic](https://www.chromatic.com/docs/cypress/#5-run-chromatic)
9. Verify the test with `cy.takeSnapshot()` has a story with the UI that it would have had at that point in the test
10. Set `disableAutoSnapshot: false` globally
11. Run Cypress again
12. Run Chromatic again
13. Verify that automatic _and_ manual snapshots are all captured
